lib/moonshot/dynamic_template.rb

```diff
@@ -58,9 +58,38 @@ def validate_destination_exists
     end
 
     def validate_template(template)
-      Aws::CloudFormation::Client.new.validate_template(
-        template_body: template
-      )
+      if template.bytesize > 50_000 # Leave some margin from the 51,200 limit
```

Comment:
@rutuja810 checking if config.template_s3_bucket is set is a better condition here since s3 is used only when this config is set.